File: app/src/main/java/com/wadektech/mtihani/viewmodels/MessagesActivityViewModel.java
package com.wadektech.mtihani.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import androidx.paging.PagedList;

import com.wadektech.mtihani.repository.MtihaniRepository;
import com.wadektech.mtihani.room.Chat;

import java.util.List;

public class MessagesActivityViewModel extends ViewModel {
    private String myId,userId;
    private LiveData<PagedList<Chat>> messagesList;

    public MessagesActivityViewModel(String myId, String userId) {
        this.myId = myId;
        this.userId = userId;
        messagesList= MtihaniRepository.loadMessagesFromRoom(myId,userId);
    }

    public LiveData<PagedList<Chat>> getMessagesList() {
        return messagesList;
    }


    public void saveMessage(Chat chat) {
        MtihaniRepository.saveMessage(chat);
    }

    public void saveNewMessages(List<Chat> chats) {
        MtihaniRepository.saveNewMessages(chats);
    }

    public void sendMessageToFirebase(Chat chat) {
        MtihaniRepository.sendMessageToFirebase(chat);
    }
}

File: app/src/main/java/com/wadektech/mtihani/chat/presentation/viewmodels/MessagesActivityViewModel.kt
package com.wadektech.mtihani.chat.presentation.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.paging.PagedList
import com.wadektech.mtihani.chat.data.localDatasource.room.Chat
import com.wadektech.mtihani.chat.data.repository.ChatsRepositoryImpl



class MessagesActivityViewModel(private val myId: String, private val userId: String) :
    ViewModel() {
    val messagesList: LiveData<PagedList<Chat>>
    private val chatsRepositoryImpl = ChatsRepositoryImpl()

    init {
        messagesList = chatsRepositoryImpl.loadMessagesFromRoom(myId, userId)
    }

    fun saveMessage(chat: Chat?) {
        chatsRepositoryImpl.saveMessage(chat)
    }

    fun saveNewMessages(chats: List<Chat?>?) {
        chatsRepositoryImpl.saveNewMessages(chats)
    }

    fun sendMessageToFirebase(chat: Chat?) {
        chatsRepositoryImpl.sendMessageToFirebase(chat)
    }
}